### ISSUE TITLE
bug: Redirect route only works once

### ISSUE THREAD

**Issue by OoDeLally** [ionic-team/ionic#20552]
# Bug Report

**Ionic version:**

5.0.0

**Current behavior:**
Redirection only works once.

## Step to reproduce

Here is a toy project to reproduce the issue:
https://github.com/OoDeLally/ionic-redirect-bug

- Run the app ( `npm i && npm start`).
- Click on `Go to Page` button.
- The app moves to `/foo`, then gets redirected to `/foo/page`.
- Click on `Back` button.
- The app goes back to the main menu.
- Click on `Go to Page` button.
- EXPECTED: The app moves to `/foo`, then gets redirected to `/foo/page`.
- RESULT: Nothing happens. The redirection route gets rendered, but doesnt show anything.

**Other information:**

**Ionic info:**

```
▶ ionic info

Ionic:

   Ionic CLI       : 5.4.4 (/home/user/.nvm/versions/node/v8.16.0/lib/node_modules/ionic)
   Ionic Framework : @ionic/react 5.0.0

Utility:

   cordova-res : not installed
   native-run  : 0.2.9 (update available: 0.3.0)

System:

   NodeJS : v8.16.0 (/home/user/.nvm/versions/node/v8.16.0/bin/node)
   npm    : 6.4.1
   OS     : Linux 4.15


```

**Comment by elylucas** [comment]
Hi @OoDeLally,

Thanks for the issue and the detailed repo. I believe this is fixed and will go out shortly in the 5.0.1 release later today. If its still happening, ping me here and I'll take another look.

Thanks

**Comment by OoDeLally** [comment]
Nice, thanks!
